packages/paste-website/src/pages/components/product-switcher/index.mdx

```diff
@@ -0,0 +1,245 @@
+import {ProductEmailAPIIcon} from '@twilio-paste/icons/esm/ProductEmailAPIIcon';
+
+import {SidebarCategoryRoutes} from '../../../constants';
+import DefaultLayout from '../../../layouts/DefaultLayout';
+import {getFeature, getNavigationData} from '../../../utils/api';
+import {productSwitcherExample} from '../../../component-examples/ProductSwitcherExamples';
+
+export default DefaultLayout;
+
+export const getStaticProps = async () => {
+  const navigationData = await getNavigationData();
+  const feature = await getFeature('Product Switcher');
+  return {
+    props: {
+      data: {
+        ...packageJson,
+        ...feature,
+      },
+      navigationData,
+    },
+  };
+};
+
+<NormalizedComponentHeader
+  categoryRoute={SidebarCategoryRoutes.COMPONENTS}
+  githubUrl="https://github.com/twilio-labs/paste/tree/main/packages/paste-core/components/product-switcher"
+  storybookUrl="/?path=/story/components-product-switcher--"
+  data={props.data}
+/>
+
+---
+
+<contentwrapper>
+
+<PageAside data={mdxHeadings} />
+
+<content>
+
+<LivePreview
+  scope={{
+    ProductSwitcher,
+    ProductSwitcherButton,
+    ProductSwitcherItem,
+    useProductSwitcherState,
+    LogoTwilioIcon,
+    ProductSegmentIcon,
+    ProductFlexIcon,
+    ProductEmailAPIIcon,
+  }}
+  noInline
+  showOverflow
+>
+  {productSwitcherExample}
+</LivePreview>
+
+## Guidelines
+
+## About Product Switcher
+
+The Product Switcher is an implementation of a Menu, that is designed to show the Twilio products a user has access to and can switch between.
+
+It is designed to be used as a quick mechnism to jump between top level Twilio product applications such as Twilio Console, Segment, Sendgrid and Flex. It is ideally placed in the application [Topbar](/components/topbar).
+
+### Accessibility
+
+The Product Switcher implements a Menu and supports all the same keyboard navigation that the [Menu](/components/menu) component does.
+
+## Examples
+
+<LivePreview
+  scope={{
+    ProductSwitcher,
+    ProductSwitcherButton,
+    ProductSwitcherItem,
+    useProductSwitcherState,
+    LogoTwilioIcon,
+    ProductSegmentIcon,
+    ProductFlexIcon,
+    ProductEmailAPIIcon,
+  }}
+  noInline
+  showOverflow
+>
+  {productSwitcherExample}
+</LivePreview>
+
+---
+
+## Composition Notes
+
+The Product Switcher is made up of the following sub-components:
+
+- `ProductSwitcherButton`
+- `ProductSwitcher`
+- `ProductSwitcherItem`
+- `useProductSwitcherState`
+
+### Product Switcher Button
+
+The Product Switcher Button is a composed Icon Only Menu Button.
+
+### Product Switcher
+
+Container for the Product Switcher Menu.
+
+### Product Switcher Item
+
+A radio menu item that can either perform an action or navigate to a new url. It can show the currently selected product the user is in. Only one product can be selected at a time via the `checked` property. Ensure that the Items all belong to the same group by sharing a `name`.
```

Comment:
Is this supposed to be "or"?:

> A radio menu item that can either perform an action or navigate to a new url.

Isn't it meant to do both? Or just navigate, really.

Reply:
Not _technically_. If it navigates, it's re-rendering the DOM so no action or selection actually happens. It just influences how the initial render of the menu looks. And it can _technically_ just perform an action if that's what you want your app to do.